### ISSUE TITLE
Create new ADR for implementing GuardDuty and Config as preventive and detective security controls

### ISSUE THREAD

**Issue by rbogle** [cloud-gov/product#2328]
In order to improve our detective and preventive security controls within AWS we would like to use and implement AWS GuardDuty and AWS Config across all of our accounts. This ADR will detail how we will do that and the benefits of them. 

## Acceptance Criteria
* [x] ADR written
* [x] detail to supply SCR for adding these services to the boundry
* [x] Delete resources created for testing in gov-root
  - AWSConfigMboydTestRule EventBridge rule
  - config-topic-mboyd-test SNS topic
  - GovAggregatorMboydTest aggregator
  - Enabled rules for AWS Config
  - config-bucket-089226552368-mboyd-test s3 bucket
* [x] Delete resources created for testing in gov-prd-plat-admin
  - config-mboyd-test SNS topic
  - config-mboyd-test-alerts EventBridge rule
  - Enabled rules for AWS Config
  - config-bucket-135676904304-mboyd-test S3 bucket
--- 

## Security considerations
None

**Comment by markdboyd** [comment]
What I've learned so far about AWS config:

- AWS Config is a service that can monitor the configuration for all of the supported resources and then report if the configuration doesn't match what is expected via rules
  - For configuration that isn't compliant with your rules, a message can be sent to an SNS topic
  - From the SNS topic we could have email notifications, etc
- There are a number of AWS managed rules for expected configuration: https://docs.aws.amazon.com/config/latest/developerguide/managed-rules-by-aws-config.html

A few notes on enabling AWS Config:

- There is a "1-click" button to enable AWS Config with the recommended default settings: https://docs.aws.amazon.com/config/latest/developerguide/1-click-setup.html
- AWS Config can be used to aggregate configuration compliance reports from a set of accounts or all accounts in an organization: https://docs.aws.amazon.com/config/latest/developerguide/aggregate-data.html
  - I assume since our accounts are not in their own AWS organization, we cannot take advantage of the organization aggregation
  - Maybe we could still aggregate all of our accounts into one account? Maybe the commercial prod account?
- We could use Terraform to enable AWS Config separately in all of our accounts? https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/config_configuration_recorder

**Comment by jameshochadel** [comment]
Just cross linking with #2231 — Per the title of this issue, maybe a separate ADR is good to keep the Shield one focused, as long as we combine them into a single SCR so we can present the changes as a batch to the JAB. Or feel free to add it on to my PR to the Shield ADR!

**Comment by markdboyd** [comment]
@jameshochadel Yeah I'm not sure if you saw Rian's comment: https://github.com/cloud-gov/product/issues/2325#issuecomment-1297500558

We think it is best to keep the current ADR focused to shield + WAF, so we can keep it nicely scoped and because it's basically done. Then we can have separate ADRs for Config + GuardDuty. 

But yeah we can combine them all into one SCR once the ADRs are done

**Comment by jameshochadel** [comment]
@markdboyd Perfect, thanks — I'd missed that.

**Comment by markdboyd** [comment]
I also just found that AWS config is enabled, possibly by TTS Tech Portfolio, in our commercial root account. I can't tell yet though whether it is only auditing resources in the commercial root account, but I believe that is the case.

**Comment by rbogle** [comment]
There are a lot of details to Config service it can do detective, preventive and even automated remediative actions on resources based on rules. And you can create your own rules and actions in it. 

We do want it applied at the org level in both commercial and govcloud. This means we can do it as soon as we want in govcloud as we have our own org there but want to centrally manage it from the org root account.  We should wait on the commercial side until we have our own org there, this may come by the end of CY 22.

There are a basic set of best practice rules, but some of these may conflict with our current patterns and will generate alerts on those as but not do any harm. Last time I worked with Config they didnt have a way to opt out of rules by account...

**Comment by rbogle** [comment]
Also IIRC auditing is where the costs come from, so we want to be smart about the number and type of rules used.

**Comment by markdboyd** [comment]
@rbogle Thanks, that's helpful! I do have a few follow-up questions

>We do want it applied at the org level in both commercial and govcloud.  

I'm not sure if I understand. I thought that there was only one "organization" level, and it's the TTS organization which owns all of our accounts? So I'm not sure what applying at the org level in commercial or govcloud would mean.

Also, I'm not sure if you saw my other comments, but TTS tech portfolio appears to already have this enabled in our commercial accounts, but I can't tell what accounts they're auditing. Should we try to get a hold of tech portfolio to find out if their intent is to monitor all of our accounts or just press forward?

Or should we just move forward with enabling on all of our accounts? It looks pretty easy to implement in Terraform: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/config_configuration_recorder

**Comment by rbogle** [comment]
An aWS account can belong to only one org, but accounts are unique in each 'partition' of AWS -- govcloud and commercial cloud are 2 separate partitions, and so an org in commercial has no relationship to an org in govcloud. Accounts in earch are effectively in two completely separate clouds. I setup and moved all of our govcloud accounts into an AWS organization with the SSO work. So things like SCPs, cloudtrail, config, etc can apply to all of the govcloud accounts at once and be managed centrally.  Anything done on the commercial side will have no effect on govcloud accounts, so we will have to replicate our services and configurations between commercial and govcloud in everything we do. 

We have multiple reasons to remove ourselves from the TTS org in AWS and I don't think we should duplicate work so I'd propose waiting to apply Config in commercial until we have our own org there and can configure across all those accounts. I don't believe Tech Portfolio is actively using or monitoring config on all the commercial accounts, but even if they are anything we can get them to change or let us change now, would have to be re-done later. 

Additionally, I think, because govcloud has the bulk of the resources and workloads of cloud.gov, govcloud accounts should be taken care of first.

**Comment by markdboyd** [comment]
Now that I have admin access to our gov and commercial accounts, I can better evaluate our options here.

For AWS Config, it looks like we can aggregate all accounts with an organization pretty easily: https://docs.aws.amazon.com/config/latest/developerguide/aggregate-data.html

The question is: is it safe to use our government root account as the aggregator for all of the other gov accounts? If we didn't use the government root account, what other account would we use?

On a related note, GuardDuty can also be configured to manage multiple accounts. But [the docs for GuardDuty ](https://docs.aws.amazon.com/guardduty/latest/ug/guardduty_organizations.html) are pretty explicit in recommending that you not use your management account as the administrator:

> Not recommended to set your organization's management account as the delegated administrator.
Your organization's management account can be the delegated administrator, but this is not recommended based on AWS Security best practices following the principle of least privilege.

Setting up central management for GuardDuty may also be more complex because it requires all accounts to be in the same region:

> A delegated administrator is Regional.
Unlike AWS Organizations, GuardDuty is a Regional service. This means that GuardDuty delegated administrators, and their member accounts must be added in each desired Region for account management through AWS Organizations to be active in every Region. In other words, if the organization management account designates a delegated administrator for GuardDuty in only US East (N. Virginia) that delegated administrator will only manage member accounts added in that Region. For more information on Regions in GuardDuty see [Regions and endpoints](https://docs.aws.amazon.com/guardduty/latest/ug/guardduty_regions.html).

**Comment by markdboyd** [comment]
SCR created: https://docs.google.com/document/d/1sfkLUhDtJYKgkH_dylhZbaShNGkXanaOWoCpPVY6a4Q/edit

**Comment by markdboyd** [comment]
The ADR and SCR have been created so closing this ticket. Will create another ticket for revising and submitting the SCR
